test_runner/src/main/kotlin/ftl/run/DumpShards.kt

```diff
@@ -12,31 +13,31 @@ import ftl.util.obfuscatePrettyPrinter
 import java.nio.file.Files
 import java.nio.file.Paths
 
-suspend fun dumpShards(
-    args: AndroidArgs,
+suspend fun AndroidArgs.dumpShards(
+    @VisibleForTesting
     shardFilePath: String = ANDROID_SHARD_FILE,
 ) {
-    if (!args.isInstrumentationTest) throw FlankConfigurationError(
+    if (!isInstrumentationTest) throw FlankConfigurationError(
         "Cannot dump shards for non instrumentation test, ensure test apk has been set."
     )
-    val shards: AndroidMatrixTestShards = args.getAndroidMatrixShards()
+    val shards: AndroidMatrixTestShards = getAndroidMatrixShards()
     saveShardChunks(
         shardFilePath = shardFilePath,
         shards = shards,
         size = shards.flatMap { it.value.shards.values }.count(),
-        obfuscatedOutput = args.obfuscateDumpShards
+        obfuscatedOutput = obfuscateDumpShards
     )
 }
```

Comment:
How about

```kotlin
getAndroidMatrixShards().run {
    saveShardChunks(
        shardFilePath = shardFilePath,
        shards = this,
        size = flatMap { it.value.shards.values }.count(),
       obfuscatedOutput = obfuscateDumpShards
    )
}
```

?

Reply:
I don't like using `val` but also I don't like unnecessary nested scopes and explicit `this` keyword so it is the same for me but the current version gives fewer changes in diff.

Reply:
I don't like using val but also I don't like unnecessary nested scopes and explicit this keyword so it is the same for me but the current version gives fewer changes in diff.